File: src/PhotoVote/Event/Candidate/CandidateImageChanged.py
from typing import Optional

from PhotoVote.Event import Event


class CandidateImageChanged(Event):
    candidate_id: Optional[str] = None
    competition_id: Optional[str] = None
    election_id: Optional[str] = None
    url: Optional[str] = None
    caption: Optional[str] = None

File: src/client/src/PhotoVote/Events/Candidate/CandidateImageChanged.ts
import {Event} from '../Event'
import {randomUUID} from "node:crypto";

class CandidateImageChanged extends Event {
    candidateId?: string
    competitionId?: string
    electionId?: string
    imageUrl?: string
    imageCaption?: string

    constructor() {
        super(randomUUID());
    }
}

export { CandidateImageChanged }